### ISSUE TITLE
"Inconsistent assumptions over interface Options" when compiling other projects

### ISSUE THREAD

**Issue by Niols** [colis-anr/morbig#37]
Compiling an other project (in this case, `shstats`) that also has an Options module leads to an error:

```
Error: The files /path/to/libmorbig/libmorbig.cmi
       and options.cmi make inconsistent assumptions over interface Options
```

This has not always been the case. I bisected this bug and found out that the last known good commit was 245208c and that the first bad commit is 0ca98a7. Sadly, there are two commits in between that do not compile, and 0ca98a7 is a very large commit, making it hard to find what went wrong.

I did not understand quite well what is causing the issue in shstats, and I didn't manage to produce a minimal example (although I'm still working on it). But it probably should not happen in any way in my opinion.

I will try to find more information about that and I keep you posted.

**Comment by Niols** [comment]
Nevermind, I found a minimal example: https://gist.github.com/Niols/9cea702c451d108b9bef5c535061b156

Using both the local Options module and anything in Libmorbig (even a type in Libmorbig.CST, as in the example) makes ocamlfind complain. I probably has to do with the way Libmorbig exports its submodules. I guess it makes the name "Options" available to anyone, which then clashes with the local module. (I wish we were using Dune.)

**Comment by yurug** [comment]
Fixed by 49f12e5...
